web-dapp/server-lib/buildSignature.js

```diff
@@ -0,0 +1,23 @@
+const sign = require('./sign');
+const Web3 = require('web3');
+
+function buildSignature(params, privateKey) {
```

Comment:
I think we should have interface similar to `sign`: here `privateKey` is mandatory, in `sign` it's optional (defaults to `config.signer_private_key`). Probably better to have it optional in both places

Reply:
The reason to make it optional in `sign` was to make it backward-compatible. `buildSignature`, on the other hand, was a new function so there was nothing to be compatible with.

I agree that making it consistent is better, though. And there is only one usage of `sign` as far as I know, so it shouldn't be riski. I will make that change.

Reply:
There are actually two places where it's used
https://github.com/poanetwork/poa-popa/blob/master/web-dapp/routes/prepare_con_tx.js#L71

https://github.com/poanetwork/poa-popa/blob/master/web-dapp/routes/prepare_reg_tx.js#L156

Reply:
Awesome, thanks.

Reply:
@phahulin Updated.